codex/blocktype.nim

```diff
@@ -31,7 +31,7 @@ import ./codextypes
 export errors, logutils, units, codextypes
 
 type
-  Block* = ref object of RootObj
+  Block* = object of RootObj
```

Comment:
I am concerned about this change. Changing Block to an object will incur a lot more copies and memory presusre since `seq[byte]` will be copied anytime the Block is copied.

Reply:
Yeah, as discussed in DMs, this is for safety and can be reverted once we move on to optimize things - but before that, we should actually measure how big the penalty is. 

The specific change was made to avoid some weirdness with ref/options/questionable.